presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -1169,4 +1176,29 @@ private static TypeSignature translateHiveUnsupportedTypeSignatureForTemporaryTa
 
         return typeSignature;
     }
+
+    public static <T> byte[] serializeZstdCompressed(Codec<T> codec, T instance)
+    {
+        try (ByteArrayOutputStream output = new ByteArrayOutputStream();
+                ZstdOutputStreamNoFinalizer zstdOutput = new ZstdOutputStreamNoFinalizer(output)) {
+            codec.writeBytes(zstdOutput, instance);
+            zstdOutput.close();
+            output.close();
+            return output.toByteArray();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
```

Comment:
```suggestion
        try (ByteArrayOutputStream output = new ByteArrayOutputStream()) {
            try (ZstdOutputStreamNoFinalizer zstdOutput = new ZstdOutputStreamNoFinalizer(output)) {
                codec.writeBytes(zstdOutput, instance);
            }
            return output.toByteArray();
        }
        catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }
```

Reply:
In theory it should be the same. Java guarantees to close all resources in a reverse order.

```
public class Main
{
    private static class Closeable1
            implements Closeable
    {

        @Override
        public void close()
        {
            System.out.println("Close Closeable1");
        }
    }

    private static class Closeable2
            implements Closeable
    {

        @Override
        public void close()
        {
            System.out.println("Close Closeable2");
        }
    }

    public static void main(String[] args)
    {
        try (Closeable1 closeable1 = new Closeable1(); Closeable2 closeable2 = new Closeable2()) {
            System.out.println("Body");
        }
    }
}
```

Prints

```
Body
Close Closeable2
Close Closeable1
```

Reply:
The comment was more around not calling close() explicitly, since the try-with-resources does that.

Reply:
Oh, sorry. I misunderstood. Let me create a patch

Reply:
Actually I'm not sure if it is correct to call `return output.toByteArray()` before the `ByteArrayOutputStream` is closed? The implementation allows it, but I wonder if that's what is expected?